src/common_routines.hpp

```diff
@@ -0,0 +1,15 @@
+#ifndef IRODS_S3_API_COMMON_ROUTINES_HPP
+#define IRODS_S3_API_COMMON_ROUTINES_HPP
+
+namespace irods::s3::api::common_routines {
+
+
+    inline std::string convert_time_t_to_str(time_t t, const std::string& format) {
+        // example 2009-10-12T17:50:30.000Z
+        char buffer[80];
+        strftime(buffer, 80, format.c_str(), localtime(&t));
+        return buffer;
```

Comment:
Could we use libfmt here? I might be missing the context.

Reply:
Seems you may be able to do that. I'm sure we've use it in the iRODS server somewhere.

See https://fmt.dev/8.1.1/api.html#date-and-time-formatting.

Up to you guys.

Reply:
Shiny!

Reply:
You can get around the compile-time issues by using `fmt::runtime(...)`. For example:
```cpp
const auto s = fmt::format(fmt::runtime("some format string {}"), 42);
```

Reply:
That is much better.  I'll make that change.

Reply:
I made the change to use fmt::runtime() and also adding back in the common_routines.hpp.  Please take a look.

I force pushed over the last commit so that the last one now will be clean.

Reply:
The change looks good.

If @alanking agrees, feel free to mark this as resolved.

Reply:
Yep, looks much better. Resolve it!